### PR TITLE
migration: create record relations

### PR DESCRIPTION
**Pull request by kprzerwa**

### CODE REVIEW

File: cds_ils/migrator/documents/api.py [hunk 1]
```diff
@@ -160,3 +160,14 @@ def get_documents_with_external_eitems():
         ],
     )
     return search
+
+
+def search_documents_with_siblings_relations():
+    """Return documents with siblings relations."""
+    search = DocumentSearch().filter(
```
Comment (ntarocco) [inline cds_ils/migrator/documents/api.py:167]: hardcoded search class

File: cds_ils/migrator/series/api.py [hunk 2]
```diff
@@ -71,124 +72,45 @@ def create_multipart_volumes(
...
-                        document["volume"],
-                    )
-        except StopIteration:
-            print('No matching could be done')
+        provider = DocumentIdProvider.create(
+            object_type="rec", object_uuid=record_uuid
+        )
+        temp["pid"] = provider.pid.pid_value
+        record = Document.create(temp, record_uuid)
+        record.commit()
```
Comment (ntarocco) [inline cds_ils/migrator/series/api.py:112]: probably `commit` not needed here

File: cds_ils/migrator/series/api.py [hunk 3]
```diff
@@ -71,124 +72,45 @@ def create_multipart_volumes(
...
-        multipart_legacy_recid
-    inherited_metadata["authors"] = \
-        inherited_metadata["_migration"]["authors"] \
-        if "authors" in inherited_metadata["_migration"] else []
-    inherited_metadata["serial_title"] = inherited_metadata.get("title")
-
+    # Re-use the current record for the first volume
+    # TODO review this - there are more cases of multiparts
+    first_volume = next(volume_numbers)
+    first = Document.get_record_by_pid(pid)
```
Comment (ntarocco) [inline cds_ils/migrator/series/api.py:90]: hardcoded class name

File: cds_ils/migrator/series/api.py [hunk 4]
```diff
@@ -71,124 +72,45 @@ def create_multipart_volumes(
...
-                        MULTIPART_MONOGRAPH_RELATION,
-                        document["volume"],
-                    )
-        except StopIteration:
-            print('No matching could be done')
+        provider = DocumentIdProvider.create(
+            object_type="rec", object_uuid=record_uuid
+        )
+        temp["pid"] = provider.pid.pid_value
+        record = Document.create(temp, record_uuid)
```
Comment (ntarocco) [inline cds_ils/migrator/series/api.py:111]: hardcoded class name

File: cds_ils/migrator/series/api.py [hunk 5]
```diff
@@ -71,124 +72,45 @@ def create_multipart_volumes(
...
+    first.commit()
+    yield first
 
     # Create new records for the rest
     for number in volume_numbers:
-        volume = volumes[number]
-        temp = inherited_metadata.copy()
-        if "title" in volume and volume["title"]:
-            temp["title"] = volume["title"]
+        temp = first.copy()
```
Comment (ntarocco) [inline cds_ils/migrator/series/api.py:103]: why the `copy`? isn't it maybe better to deep clone the dict?

File: cds_ils/migrator/series/api.py (hunk 2 above)
Comment (kprzerwa) [reply cds_ils/migrator/series/api.py:112]: the volume was not saved without it

**Review by jrcastro2 — APPROVED** [review]

**Pull request merged**